### PR TITLE
Feat/disclosure

### PR DESCRIPTION
**Pull request by marianayovcheva**
Added the disclosure component which already exists in public-website. This addresses Issue #1718.
I have added a couple of question I have about BDD - and that section will generally require more attention when reviewing.

### CODE REVIEW

File: testing/features/support/components/disclosure/example.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+module Disclosure
+  class Example < ::Base
+    set_url "/iframe.html?id=components-disclosure--example&args=&viewMode=story"
+
+    element :toggle, ".cads-disclosure__toggle"
```
Comment (marianayovcheva) [inline testing/features/support/components/disclosure/example.rb:7]: Not sure if it's better for me to add `data-test-id` instead

File: testing/features/components/step_definitions/disclosure_steps.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+Given("the Example Disclosure component is on the page") do
+  @component = Disclosure::Example.new.tap(&:load)
+end
+
+When("I open/close the disclosure component") do
```
Comment (marianayovcheva) [inline testing/features/components/step_definitions/disclosure_steps.rb:7]: This works but I don't fully get how it works - i.e how open/close can then be translated to two expressions later (i.e `I open the disclosure component` and `I close the disclosure component`. Links to docs would be greatly appreciated - I've tried to look myself, but I'm probably not wording the search term properly.

File: testing/features/components/step_definitions/disclosure_steps.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  @component.toggle.click
+end
+
+Then("The disclosure component displays expanded content") do
```
Comment (marianayovcheva) [inline testing/features/components/step_definitions/disclosure_steps.rb:11]: For those two I've tried to match what's in [public-website](https://github.com/citizensadvice/public-website/blob/develop/features/advice/public/step_definitions/search_steps.rb#L96). The problem I was having was that when I was using `.not_to be_visible` was throwing an error:
image
What I have now works, however I want to understand why `.not_to be_visible` was not working in this case

File: testing/features/components/step_definitions/disclosure_steps.rb (hunk 2 above)
Comment (davidrapson) [reply testing/features/components/step_definitions/disclosure_steps.rb:7]: These are cucumber expressions. The only docs I could find are the Java one but the syntax is the same https://github.com/cucumber/cucumber-expressions#readme

File: testing/features/support/components/disclosure/example.rb (hunk 1 above)
Comment (davidrapson) [reply testing/features/support/components/disclosure/example.rb:7]: I don't mind the classes for design system components as class names are ultimately part of the public API. `data-testid` would be fine too.

**Review by davidrapson — APPROVED** [review]
💯 This looks fantastic

**Pull request merged**